rest-catalog/quarkus-spark-iceberg-tests/src/intTest/java/org/projectnessie/restcatalog/intttests/ITNessieCatalogServerSparkSQL.java

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright (C) 2023 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.restcatalog.intttests;
+
+import static java.util.Collections.emptyMap;
+import static java.util.Objects.requireNonNull;
+import static org.projectnessie.testing.keycloak.CustomKeycloakContainer.CLIENT_SECRET;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.iceberg.CatalogProperties;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.rest.RESTCatalog;
+import org.apache.iceberg.rest.auth.OAuth2Properties;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.keycloak.representations.idm.RealmRepresentation;
+import org.keycloak.representations.idm.RoleRepresentation;
+import org.projectnessie.client.NessieConfigConstants;
+import org.projectnessie.client.http.HttpClientBuilder;
+import org.projectnessie.nessierunner.common.ProcessHandler;
+import org.projectnessie.spark.extensions.AbstractNessieSparkSqlExtensionTest;
+import org.projectnessie.testing.keycloak.CustomKeycloakContainer;
+import org.projectnessie.testing.nessie.NessieContainer;
+import org.testcontainers.containers.Network;
+
+public class ITNessieCatalogServerSparkSQL extends AbstractNessieSparkSqlExtensionTest {
```

Comment:
Some tests are failing:

* `mergeReferencesIn`
* `mergeReferencesIntoMain`

```
Expecting actual:
  [SparkCommitLogEntry{author=, committer=, signedOffBy=, properties={test=123}}]
to contain exactly (and in same order):
  [SparkCommitLogEntry{author=, committer=, signedOffBy=, properties={}}]
but some elements were not found:
  [SparkCommitLogEntry{author=, committer=, signedOffBy=, properties={}}]
and others were not expected:
  [SparkCommitLogEntry{author=, committer=, signedOffBy=, properties={test=123}}]
```

Reply:
I don't see those failing - neither locally nor in CI?

Reply:
They fail consistently for me. Same error.

Reply:
Works on all machines (Linux + macOS) here. Can you look why that fails for you?

Reply:
This is driving me nuts...

The issue is with the merge operation.

When running `ITNessieCatalogSparkSQL.mergeReferencesIntoMain`, the entry `test=123` does not get included in the merge commit:

<img width="925" alt="image" src="https://github.com/projectnessie/nessie/assets/463876/dfc69b0c-33a0-45c0-9968-185c72cdab67">

But when running `ITNessieCatalogServerSparkSQLmergeReferencesIntoMain`, the entry `test=123` is included in the merge commit:

<img width="1005" alt="image" src="https://github.com/projectnessie/nessie/assets/463876/d95ab8f4-2349-49d7-89fc-b0b0a1208a3a">

Which is mind-boggling, since in `BaseMergeBranchExec` I don't see any `CommitMeta` being sent to the server, so the commit merge should not have any properties.

Reply:
Oh I'm starting to understand: it depends on the Nessie version.

It works for Nessie > 0.61, it fails otherwise. The latest tag on my machine was likely pointing to an old version.

Reply:
The "culprit" is likely https://github.com/projectnessie/nessie/pull/7035.

Reply:
> Pulling a newer image fixed the issue

;) ;)